backend/src/middleware/file_upload_security.py

```
@@ -0,0 +1,694 @@
+
+        return result
+
+    def _validate_mime_type(self, filename: str, content: bytes) -> Dict[str, Any]:
+        """Validate MIME type using magic bytes"""
+        result = {'errors': [], 'warnings': [], 'detected_mime': None}
+
+        # Detect MIME type from content
+        try:
+            detected_mime = magic.from_buffer(content, mime=True)
+            result['detected_mime'] = detected_mime
+        except Exception as e:
```

Ruff (0.14.0)

Do not catch blind exception: `Exception`

(BLE001)